lib/Agrammon/Model.pm6

```diff
@@ -156,30 +158,32 @@ class Agrammon::Model {
         method !annotate-inputs-internal($input-data, %run-already, @result, Str $instance-id?) {
             my $tax = $!module.taxonomy;
             return if %run-already{$tax};
+            my $gui-root = $!module.gui-root-module;
             if $!module.is-multi {
                 for $input-data.inputs-list-for($tax) -> $multi-input {
                     my $instance-id = $multi-input.instance-id;
                     self!annotate-inputs-as-single($multi-input, %run-already.clone,
-                            @result, $instance-id);
+                            @result, $gui-root, $instance-id);
                 }
                 self!mark-multi-run(%run-already);
             }
             else {
-                self!annotate-inputs-as-single($input-data, %run-already, @result, $instance-id);
+                self!annotate-inputs-as-single($input-data, %run-already, @result, $gui-root, $instance-id);
                 %run-already{$tax} = True;
             }
         }
 
-        method !annotate-inputs-as-single($input-data, %run-already, @result, Str $instance-id?) {
+        method !annotate-inputs-as-single($input-data, %run-already, @result, $gui-root, Str $instance-id?) {
             for @!dependencies -> $dep {
                 $dep!annotate-inputs-internal($input-data, %run-already, @result, $instance-id);
             }
             my %input-data := $input-data.input-hash-for($!module.taxonomy);
             my %input-defaults := $!module.input-defaults;
             for $!module.input -> Agrammon::Model::Input $input {
-                my $key = $input.name;
+                my $key   = $input.name;
                 my $value = %input-data{$key} // %input-defaults{$key};
-                @result.push: AnnotatedInput.new: :$!module, :$input, :$instance-id, :$value;
+                my $enum  = $input.enum;
+                @result.push: AnnotatedInput.new: :$!module, :$input, :$instance-id, :$value, :$enum, :$gui-root;
```

Comment:
I'm not sure why we add `enum` as an attribute on the class. It's already accessible via `input`, which is also attached to the class. If you simply want the convenience, just do `has Agrammon::Model::Input $.input is required handles <enum>;` and you'll get a delegating method installed.

Reply:
Defnitely not needed, removed.